velox/functions/prestosql/Slice.cpp

```diff
@@ -200,22 +204,36 @@ class SliceFunction : public exec::VectorFunction {
     return std::min(endIndex - start, length);
   }
 };
+} // namespace
+
+template <TypeKind IndexKind>
+void registerSliceFunction(const std::string& prefix) {
+  using NativeType = typename TypeTraits<IndexKind>::NativeType;
+  auto indexKindName = TypeTraits<IndexKind>::name;
+  std::vector<std::shared_ptr<exec::FunctionSignature>> signatures;
+  if constexpr (
+      std::is_same_v<NativeType, int64_t> ||
+      std::is_same_v<NativeType, int32_t>) {
+    signatures.push_back(exec::FunctionSignatureBuilder()
+                             .typeVariable("T")
+                             .returnType("array(T)")
+                             .argumentType("array(T)")
+                             .argumentType(indexKindName)
+                             .argumentType(indexKindName)
+                             .build());
+  } else {
+    VELOX_FAIL("Unsupported index type {} to register slice()", indexKindName);
```

Comment:
nit: a message suggestion: Unsupported parameter type {} to register slice function.